lua/nvim-tree/explorer/init.lua

```diff
@@ -683,6 +686,102 @@ function Explorer:get_nodes()
   return self:clone()
 end
 
+---@param new_tabpage integer
+---@return boolean
+function Explorer:is_window_event(new_tabpage)
+  local is_event_scope_window = vim.v.event.scope == "window" or vim.v.event.changed_window or false
+  return is_event_scope_window and new_tabpage == self.current_tab
+end
+
+---@param name string
+---@return string|nil
+function Explorer:clean_input_cwd(name)
+  name = vim.fn.fnameescape(name)
+  local cwd = core.get_cwd()
+  if cwd == nil then
+    return
+  end
+  local root_parent_cwd = vim.fn.fnamemodify(utils.path_remove_trailing(cwd), ":h")
+  if name == ".." and root_parent_cwd then
+    return vim.fn.expand(root_parent_cwd)
+  else
+    return vim.fn.expand(name)
+  end
+end
+
+---@param foldername string
+---@return boolean
+function Explorer:prevent_cwd_change(foldername)
+  local is_same_cwd = foldername == core.get_cwd()
+  local is_restricted_above = config.restrict_above_cwd and foldername < vim.fn.getcwd(-1, -1)
```

Comment:
`config.restrict_above_cwd` is incorrect. We need to specify the full path when we are inside Explorer: `config.actions.change_dir.restrict_above_cwd`

It worked in change-dir.lua as it set `M.options = options.actions.change_dir`